torch/_inductor/coordinate_descent_tuner.py

```diff
@@ -134,24 +236,36 @@ def autotune(
                 for next_val in candidate_values:
                     candidate_config = copy.deepcopy(best_config)
                     set_field(candidate_config, name, next_val)
-                    log.debug("Try config %s", candidate_config)
-                    try:
-                        candidate_timing = self.call_func(func, candidate_config)
-                    except Exception as e:
-                        log.debug("Got exception %s", e)
-                        continue
-
-                    if self.has_improvement(best_timing, candidate_timing):
-                        improved = True
-                        log.debug(
-                            "Tune from %s %f -> %s %f",
-                            best_config,
-                            best_timing,
-                            candidate_config,
-                            candidate_timing,
-                        )
-                        best_timing = candidate_timing
-                        best_config = candidate_config
+
+                    cmp_res, candidate_timing = self.compare_config(
+                        func, candidate_config, best_config, best_timing
+                    )
+                    if cmp_res:
+                        best_config, best_timing = candidate_config, candidate_timing
+
+            if not improved and config.coordinate_descent_check_all_dir:
+                self.best_config = best_config
+                self.best_timing = best_timing
+                self.cur_config = copy.deepcopy(self.best_config)
+
+                improved = self.check_all_dir(func)
+
+                if improved:
+                    import colorama
```

Comment:
We should handle the case where this isn't installed.

Reply:
I can add a few util in util.py like 
- make_red_text , make_yellow_text
```
def make_red_text(text):
  if HAS_COLORAMA:
    return colorama.Fore.RED + text + colorama.Fore.RESET
  else:
    return text
```
they use colorama if availabe, and fallback to black color text otherwise.